mne_bids/tests/test_write.py

```diff
@@ -509,7 +509,7 @@ def test_fif(_bids_validate, tmpdir):
     raw = _read_raw_fif(raw_fname, preload=True)
     with pytest.raises(ValueError, match='preloaded'):
         write_raw_bids(raw, bids_path_meg, events_data=events,
-                       event_id=event_id, overwrite=False)
+                       event_id=event_id, preloaded=False, overwrite=False)
```

Comment:
I like `allow_preloaded` more that @agramfort suggested :)

can you update a test that checks that the error is not raised when `preload=False` and `allow_preloaded=True` ?

Reply:
+1 for allow_preload(ed)

Reply:
> can you update a test that checks that the error is not raised when `preload=False` and `allow_preload=True` ?

Would you like a simple test for no error or a full test that writes, reads, and checks that the data are equal?

Reply:
a simple test is fine, maybe with the `EvokedArray` example that you shared when reporting the issue. So we know that it works

Reply:
As you [mentioned](https://github.com/mne-tools/mne-bids/issues/803#issuecomment-855973739), looks like format should be also explicit in this case. Some filename checks also have to be skipped, since preloaded data are not necessarily tied to a file. I will add a separate test for this.